client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import { Banner } from '../index';
 
-jest.mock( 'calypso/lib/analytics/track-component-view', () => {
-	return function TrackComponentView() {
-		return null;
+jest.mock( 'calypso/state/analytics/actions', () => {
+	return {
+		recordTracksEvent: jest.fn( () => ( {
+			type: 'ANALYTICS_EVENT_RECORD',
+		} ) ),
 	};
 } );
 
-import { Card, Button } from '@automattic/components';
-import { shallow } from 'enzyme';
-import PlanPrice from 'calypso/my-sites/plan-price/';
-import { Banner } from '../index';
+function renderWithRedux( ui ) {
+	return renderWithProvider( ui, {
+		reducers: {
+			preferences: preferencesReducer,
+		},
+		initialState: { preferences: { remoteValues: {} } },
+	} );
+}
 
 const props = {
 	callToAction: null,
 	title: 'banner title',
 };
 
 describe( 'Banner basic tests', () => {
-	test( 'should not blow up and have proper CSS class', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner' ) ).toHaveLength( 1 );
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	test( 'should output Banner component with proper CSS class', () => {
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.firstChild ).toHaveClass( 'banner' );
 	} );
 
 	test( 'should render Card if dismissPreferenceName is null', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ null } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } dismissPreferenceName={ null } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).not.toHaveClass( 'is-dismissible' );
 	} );
 
 	test( 'should render DismissibleCard if dismissPreferenceName is defined', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
-		expect( comp.find( Card ) ).toHaveLength( 0 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 1 );
+		renderWithRedux( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
+		expect( screen.getByLabelText( 'Dismiss' ) ).toBeInTheDocument();
 	} );
 
 	test( 'should have .has-call-to-action class if callToAction is defined', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
+		expect( container.firstChild ).toHaveClass( 'has-call-to-action' );
 	} );
 
 	test( 'should not have .has-call-to-action class if callToAction is null', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ null } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } callToAction={ null } /> );
```

Comment:
We might as well not provide the prop, as it seems to be declared inside the `props` already? Also, I don't think it would make any difference if we remove it from `props` completely too in this scenario.

```suggestion
		const { container } = render( <Banner { ...props } /> );
```

Reply:
In that case, it seems like it makes sense to remove this test completely since there is already another test `'should not render a <Button /> when callToAction is not specified'` which would be a duplicate of what you're suggesting.

Reply:
That would make sense to me 👍 I guess this tests the default prop value, which if we change, would break both tests anyway.